distro.py

```diff
@@ -960,6 +967,18 @@ def _parse_lsb_release_content(lines):
             props.update({k.replace(' ', '_').lower(): v.strip()})
         return props
 
+    @staticmethod
+    def _parse_uname_content(lines):
```

Comment:
2 extra pieces of information could be added for FreeBSD:

```
$ uname -rs
FreeBSD 11.1-RELEASE-p4
```

1. On FreeBSD the distro.codename() function could output either "RELEASE", "STABLE" or "CURRENT" which are described here: https://www.freebsd.org/doc/handbook/current-stable.html

1. On FreeBSD the distro.build_number() output could be the kernel patch level. FreeBSD releases get regular security/errata patches which get added to the uname output if they affect the kernel. In the above output, the kernel is at patch level "p4". Note that the userland can at times be patched separate from the kernel, leading to userland patchlevel and kernel patchlevel being different. Note that uname only provides kernel details. The way to show userland patch level on FreeBSD is "freebsd-version -u", whilst kernel patch level is shown on uname or exactly the same in "freebsd-version -k".

Reply:
1. In my VM I didn't see the `-p4`, only the `-RELEASE`. The thing is that's not really the "codename"? Codename typically is the name of the release like "Trusty" for Ubuntu 14.04. I'm not sure if `RELEASE`, `STABLE` and `CURRENT` fit that bill enough to use them for `distro.codename()`?

2. Could the `p` be dropped from the `p4`, does it have meaning? I'm pretty sure no other `distro.version()` call will result in anything but integers and dots so it would break `distro.version_parts()`'s contract? Not 100% certain here.